backend/api/Services/MissionRunService.cs

```diff
@@ -118,6 +120,14 @@ public async Task<PagedList<MissionRun>> ReadAll(MissionRunQueryStringParameters
                 .FirstOrDefaultAsync(missionRun => missionRun.Id.Equals(id));
         }
 
+        public async Task<MissionRun?> ReadNextScheduledRunByMissionId(string missionId)
+        {
+            return await GetMissionRunsWithSubModels()
+                .Where(m => m.MissionId == missionId && m.EndTime == null)
```

Comment:
```suggestion
                .Where(m => m.MissionId == missionId && !m.IsCompleted)
```
Would this be more legible?

Reply:
I agree that it would be more legible, but since IsCompleted is evaluated locally and does not exist in the database, we cannot put it in the LINQ statement. We could however check the state and make sure that it is cancelled, failed, successful or partially successful, but then it becomes less legible again in my opinion.